### PR TITLE
Seanquiambao/admin page

### LINKED ISSUE #206: Code Admin Page
image

### PR DESCRIPTION
**Pull request by seanquiambao**
The page is pretty much finished but the page.js is in the wrong directory, so the URL is in /admin/admin. Not sure if I should put it in the appropriate directory now or after issue #205 gets resolved. Other than that, let me know if there's anything I need to change!

adminpage

- Closes issue #206

### CODE REVIEW

File: src/components/Admin/Tag.jsx [hunk 1]
```diff
@@ -40,6 +40,22 @@ const colors = {
     background: "bg-hackathon-judge-student-bg",
     text: "text-hackathon-judge-student-text",
   },
+  marketing: {
```
Comment (shahdivyank) [inline src/components/Admin/Tag.jsx]: make these color based not role based

File: src/components/Admin/Admins.jsx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    affiliation: "off",
+    status: "off",
+  });
+
+  const tags = [
+    {
+      color: "green",
+      text: "accept",
+      name: "Accept",
+      onClick: () => {
```
Comment (shahdivyank) [inline src/components/Admin/Admins.jsx]: these should be populated with the proper functions, checkout the judges page for code

File: src/components/Admin/Admins.jsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+      },
+    },
+    {
+      color: "red",
+      text: "remove",
+      name: "Remove",
+      onClick: (setToggle) => {
+        setToggle(false);
+        setFilteredAdmins(
+          filteredAdmins.filter((a) => {
```
Comment (shahdivyank) [inline src/components/Admin/Admins.jsx:103]: this should only be updating status, not removing them

File: src/components/Admin/Admins.jsx [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+              {header.name}
+              {header.icon && (
+                <SortIcon
+                  name={header.name.toLowerCase()}
+                  sorts={sorts}
+                  setSorts={setSorts}
+                  setfilteredObjects={setFilteredAdmins}
+                  objects={filteredAdmins}
+                  reset={{
+                    name: "off",
```
Comment (shahdivyank) [inline src/components/Admin/Admins.jsx]: this reset doesn't match with the sorts u have at the top. Make a variable that stores this reset and pass it into the use state as well as here. check the other pages for reference

File: src/components/Admin/Toolbar.jsx [hunk 5]
```diff
@@ -44,7 +44,10 @@ const Toolbar = ({
         let boolean = false;
 
         Object.keys(filters).map((value) => {
-          if (a.status === value && filters[value]) {
+          if (
```
Comment (shahdivyank) [inline src/components/Admin/Toolbar.jsx]: remove the affiliation. for the filters, we will be changing them to pending, approved, rejected as the other pages as well

File: src/components/Admin/Filters.jsx [hunk 6]
```diff
@@ -19,7 +19,7 @@ const Filters = ({
 
         Object.keys(filterValues).map((value) => {
           if (
-            a.status === value &&
+            (a.status === value || a.affiliation === value) &&
```
Comment (shahdivyank) [inline src/components/Admin/Filters.jsx]: remove affiliation, filtering will only work with status

**Review by shahdivyank — CHANGES_REQUESTED** [review]

File: src/components/Admin/Tag.jsx (hunk 1 above)
Comment (seanquiambao) [reply src/components/Admin/Tag.jsx]: not sure if this is what you meant, but i changed the admin.jsx tag to use the judges colors instead.

File: src/app/admin/admin/page.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import ProtectedPage from "@/components/ProtectedPage";
+import Admins from "@/components/Admin/Admins";
+
+const AdminPage = () => {
+  return (
+    <ProtectedPage title="Admin" restrictions={["admin"]}>
+      <div className="w-11/12">
+        <title>Admin</title>
```
Comment (shahdivyank) [inline src/app/admin/admin/page.js]: u don't need this, this is integrated alr within protected pages

**Review by shahdivyank — CHANGES_REQUESTED** [review]

**Review by shahdivyank — APPROVED** [review]

**Pull request merged**